pkg/kubectl/rolling_updater.go

```diff
@@ -306,8 +306,9 @@ func (r *RollingUpdater) scaleDown(newRc, oldRc *api.ReplicationController, desi
 	}
 	// The old controller is considered as part of the total because we want to
 	// maintain minimum availability even with a volatile old controller.
-	// Scale down as much as possible while maintaining minimum availability.
-	decrement := (oldAvailable + newAvailable) - minAvailable
+	// Scale down as much as possible while maintaining minimum availability
+	unavailableOldReplicas := oldRc.Spec.Replicas - oldAvailable
+	decrement := unavailableOldReplicas + (oldAvailable + newAvailable) - minAvailable
```

Comment:
Should this just be `oldRc.Spec.Replicas + newAvailable - minAvailable`? The math is the same.